store/src/java/com/zimbra/cs/service/mail/ToXML.java

```diff
@@ -1010,6 +1013,44 @@ public static Element encodeTag(Element parent, ItemIdFormatter ifmt, OperationC
         return encodeTag(parent, ifmt, octxt, tag, NOTIFY_FIELDS);
     }
 
+    public static Element encodeSmartFolder(Element parent, ItemIdFormatter ifmt, OperationContext octxt, SmartFolder smartFolder, int fields)
+    throws ServiceException {
+        Element el = parent.addNonUniqueElement(MailConstants.E_SMART_FOLDER);
+        el.addAttribute(MailConstants.A_ID, ifmt.formatItemId(smartFolder));
```

Comment:
Can some of this code be shared with Tag?

Reply:
Probably, I just tried to avoid making too many changes to the mess that is the `ToXML` class :)